File: Watcher.DAL.Database/Models/TestExecution.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace Watcher.DAL.Database.Models
{
    public class TestExecution: IdAware
    {
        [Required]
        public DateTime DateTime { get; set; }

        [Required]
        public string Log { get; set; }

        [Required]
        public bool IsSuccessful { get; set; }

        [Required]
        public int TestId { get; set; }

        public Test Test { get; set; }
    }
}

File: client_app/packages/watcher_client_bll/lib/models/test_execution/test_execution.dart
class TestExecution {
  final int id;
  final DateTime dateTime;
  final String log;
  final int testId;
  final bool isSuccessful;

  TestExecution({
    required this.id,
    required this.dateTime,
    required this.log,
    required this.testId,
    required this.isSuccessful
  });
}
